spring-kafka/src/main/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactory.java

```diff
@@ -185,11 +185,39 @@ private TopicPartition resolveDestination(ConsumerRecord<?, ?> cr, Exception e)
 				? "none"
 				: nextDestination.getDestinationName()));
 
+		logException(e, cr, nextDestination);
+
 		return nextDestination.isNoOpsTopic()
 					? null
 					: resolveTopicPartition(cr, nextDestination);
 	}
 
+	private void logException(Exception e, ConsumerRecord<?, ?> cr, DestinationTopic nextDestination) {
+		if (nextDestination.isDltTopic()) {
+			LOGGER.error(e, () -> getErrorMessage(cr)
+					+ "Sending to DLT with name " + nextDestination.getDestinationName() + ".");
+		}
+		else if (nextDestination.isNoOpsTopic()) {
```

Comment:
"no ops" ? In the public API?

Doesn't sound neutral and technical.

How about `NONE` instead?

Not related, though, but I don't see any JavaDocs on those public APIs of the `DestinationTopic`.
At least that `Type` enums should have some explanations for their reason. 
Maybe then I wouldn't complain about "no ops" 😉

Reply:
Well, I remember I had to figure out a way to mark `end of line`, and at the time that seemed the best approach. I don't really remember which feature needed this instead of for example returning `null` as the next destination for an input topic - I can probably figure it out by reviewing the code.

Also, TBH I never thought of this as `public API` back then, though now of course I realize it is.

So I probably should write the javadocs to these methods and enums, and then we can evaluate if we need to change anything. But probably not as part of this PR, right?